api/grpcserver/grpc.go

```diff
@@ -0,0 +1,74 @@
+package grpcserver
+
+import (
+	"fmt"
+	"github.com/spacemeshos/go-spacemesh/log"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/keepalive"
+	"google.golang.org/grpc/reflection"
+	"net"
+	"time"
+)
+
+// ServiceAPI allows individual grpc services to register the grpc server
+type ServiceAPI interface {
+	RegisterService(*Server)
+}
+
+// Server is a very basic grpc server
+type Server struct {
+	Port       int
+	GrpcServer *grpc.Server
+}
+
+// NewServer creates and returns a new Server
+func NewServer(port int) *Server {
+	return &Server{
+		Port:       port,
+		GrpcServer: grpc.NewServer(ServerOptions...),
+	}
+}
+
+// Start starts the server
+func (s *Server) Start() {
+	log.Info("starting new grpc server")
+	go s.startInternal()
```

Comment:
what if start is called twice?

Reply:
If it's called a second time on the same port, it will fail with an error about the port being unavailable. If it's called with a different port, I'd argue that's fine - the node operator might want to run two different sets of services on different ports or interfaces, right? (This is not possible if you run the app the normal way, from the commandline, of course. It would require custom code.)